src/daft-dsl/src/resolve_expr/mod.rs

```diff
@@ -217,19 +217,24 @@ fn expand_wildcards(
 /// - sum(col("a"))
 /// - sum(col("a")) > 0
 /// - sum(col("a")) - sum(col("b")) > sum(col("c"))
+/// - sum(col("a")) + col("b") when "b" is a group by key
 ///
 /// Not allowed:
-/// - col("a")
+/// - col("a") when "a" is not a group by key
 ///     - not an aggregation
-/// - sum(col("a")) + col("b")
-///     - not all branches are aggregations
-fn has_single_agg_layer(expr: &ExprRef) -> bool {
-    match expr.as_ref() {
-        Expr::Agg(agg_expr) => !agg_expr.children().iter().any(has_agg),
-        Expr::Column(_) => false,
-        Expr::Literal(_) => true,
-        _ => expr.children().iter().all(has_single_agg_layer),
-    }
+/// - sum(col("a")) + col("b") when "b" is not a group by key
+///     - not all branches are aggregations, literals, or group by keys
+fn has_single_agg_layer(expr: &ExprRef, groupby: &HashSet<ExprRef>) -> bool {
```

Comment:
could we move `has_single_agg_layer` and `validate_expr_in_agg` to be a member of `ExprResolver`. It looks like they're only ever used as part of the expr resolving.